apps/server/src/modules/board/uc/submission-item.uc.ts

```diff
@@ -49,11 +49,13 @@ export class SubmissionItemUc extends BaseUc {
 
 		let submissionItems = submissionContainerElement.children.filter(isSubmissionItem);
 
-		const boardAuthorizable = await this.boardDoAuthorizableService.getBoardAuthorizable(submissionContainerElement);
-		let users = boardAuthorizable.users.filter((user) => user.userRoleEnum === UserRoleEnum.STUDENT);
+		const boardDoAuthorizable = await this.boardDoAuthorizableService.getBoardAuthorizable(submissionContainerElement);
+
+		// only board readers canto create submission items
+		let users = boardDoAuthorizable.users.filter((user) => user.roles.includes(BoardRoles.READER));
 
-		const isAuthorizedStudent = await this.isAuthorizedStudent(userId, submissionContainerElement);
-		if (isAuthorizedStudent) {
+		// board readers can only see their own submission item
+		if (this.isUserBoardReader(userId, boardDoAuthorizable.users)) {
 			submissionItems = submissionItems.filter((item) => item.userId === userId);
 			users = [];
```

Comment:
Is this the right behaviour?
I would have thought the users array would at least contain the own user.

I can see that this is even written in the tests but I think that this might be not what we really want this api to return.
The client must have additional logic to handle put back the missing user.

Reply:
this is for the response. For student, they only get to see their own submission
the list of users is used for teacher view, to create the table of each student and the data which of them has submitted the abgabe and which not. 
This part was not changed.